src/model_signing/_signing/sign_certificate.py

```diff
@@ -141,11 +142,22 @@ def __init__(
     def _verify_bundle(self, bundle: bundle_pb.Bundle) -> tuple[str, bytes]:
         public_key = self._verify_certificates(bundle.verification_material)
         envelope = bundle.dsse_envelope
-        public_key.verify(
-            envelope.signatures[0].sig,
-            sigstore_pb.pae(envelope.payload),
-            ec.ECDSA(ec_key.get_ec_key_hash(public_key)),
-        )
+        try:
+            public_key.verify(
+                envelope.signatures[0].sig,
+                sigstore_pb.pae(envelope.payload),
+                ec.ECDSA(ec_key.get_ec_key_hash(public_key)),
+            )
+        except exceptions.InvalidSignature:
+            # Compatibility layer with pre 1.0 release
+            # Here, we patch over a bug in `pae` which mixed unicode `str` and
+            # `bytes`. As a result, additional escape characters were added to
+            # the material that got signed over.
+            self._public_key.verify(
+                envelope.signatures[0].sig,
```

Comment:
is this a typo?

`self._public_key` is used here, vs `public_key` above

Reply:
This is indeed a copy-paste typo. Fixed, thanks for catching!